test/surface/formatter_test.exs

```diff
@@ -83,6 +83,13 @@ defmodule Surface.FormatterTest do
       )
     end
 
+    test "generator attributes are formatted successfully" do
+      assert_formatter_doesnt_change("""
+      <div :for={item <- @items, item.state == :valid}>
+      </div>
+      """)
+    end
```

Comment:
Thanks for adding this. Just double checking? It doesn't seem that your changes should affect whether this worked properly.

Reply:
You're correct - I added it because it occurred to me that generator expressions might have been broken more generally; and after it didn't fail helped me figure out I could just call `render_attribute` :-)